### DISCUSSION TITLE
Configure tsserver initializationOptions

### DISCUSSION THREAD

**Discussion by rogeriofrsouza** [LazyVim/LazyVim#2043] [Q&A] [answered]
Hello 👋🏻 

I would like to set some tsserver [initializationOptions](https://github.com/typescript-language-server/typescript-language-server/blob/master/docs/configuration.md#initializationoptions). Mostly interested on setting **importModuleSpecifierPreference** to "relative" and  **importModuleSpecifierEnding** to "minimal".

This config works here, but I lose LazyVim [extra typescript commands](https://www.lazyvim.org/extras/lang/typescript) (OrganizeImports and RenameFile), which I found really useful: 
```lua
{
    "neovim/nvim-lspconfig",
    config = function()
      require("lspconfig").tsserver.setup({
        init_options = {
          preferences = {
            importModuleSpecifierPreference = "relative",
            importModuleSpecifierEnding = "minimal",
          },
        },
      })
    end,
  },
```

Is there another way to configure this LSP?

**Comment by dpetka2001** [comment]
Instead of `config` function could you try the following
```lua
opts = {
  servers = {
    tsserver = {
      settings = {
        typescript = {
          preferences = {
            importModuleSpecifierPreference = "relative",
            importModuleSpecifierEnding = "minimal",
          },
        },
      },
    },
  },
}
```
and see if it works for you?

**Reply by rogeriofrsouza** [reply]
I've tried this before and still not working 😞

**Comment by rogeriofrsouza** [comment, marked as answer]
Hey, this snippet works 😄 :

```lua
{
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        tsserver = {
          init_options = {
            preferences = {
              importModuleSpecifierPreference = "relative",
              importModuleSpecifierEnding = "minimal",
            },
          },
        },
      },
    },
  },
```

Thank you @dpetka2001 for your help!
